google-cloud-bigtable/lib/google/cloud/bigtable/service.rb

```diff
@@ -279,16 +279,12 @@ def delete_cluster instance_id, cluster_id
         # @return [Google::Cloud::Bigtable::Admin::V2::Table]
         #
         def create_table instance_id, table_id, table, initial_splits: nil
-          initial_splits = initial_splits.map { |key| { key: key } } if initial_splits
+          initial_splits = Array(initial_splits).map { |key| { key: key } }
```

Comment:
This will change the behavior: `initial_splits` will become the empty array if `nil` is passed in, whereas before it would stay `nil`. As a result, we would now be passing `initial_splits: []` to `create_table` explicitly whereas we were omitting the argument earlier. Just want to make sure this change is a noop.

Reply:
Without this change, we now get this error. I will try to pinpoint the dependency change that caused it:

```
m /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable/lib/google/cloud/bigtable/project.rb:428:in `create_table'
	 8: from /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable/lib/google/cloud/bigtable/table.rb:441:in `create'
	 7: from /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable/lib/google/cloud/bigtable/service.rb:284:in `create_table'
	 6: from /Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-bigtable-admin-v2/lib/google/cloud/bigtable/admin/v2/bigtable_table_admin/client.rb:328:in `create_table'
	 5: from /Users/quartzmo/.rbenv/versions/2.7.2/gemsets/quartzmo/gems/gapic-common-0.4.0/lib/gapic/protobuf.rb:37:in `coerce'
	 4: from /Users/quartzmo/.rbenv/versions/2.7.2/gemsets/quartzmo/gems/gapic-common-0.4.0/lib/gapic/protobuf.rb:54:in `coerce_submessages'
	 3: from /Users/quartzmo/.rbenv/versions/2.7.2/gemsets/quartzmo/gems/gapic-common-0.4.0/lib/gapic/protobuf.rb:54:in `each'
	 2: from /Users/quartzmo/.rbenv/versions/2.7.2/gemsets/quartzmo/gems/gapic-common-0.4.0/lib/gapic/protobuf.rb:57:in `block in coerce_submessages'
	 1: from /Users/quartzmo/.rbenv/versions/2.7.2/gemsets/quartzmo/gems/gapic-common-0.4.0/lib/gapic/protobuf.rb:81:in `coerce_submessage'
/Users/quartzmo/.rbenv/versions/2.7.2/gemsets/quartzmo/gems/gapic-common-0.4.0/lib/gapic/protobuf.rb:99:in `coerce_array': Value  must be an array (ArgumentError)
```

Reply:
I would actually expect the original code to continue to work, as we're passing a hash (which is a valid request object type) to a generated client that can take it. If we're getting the above error, I'd expect that the original code's `delete_if` would have dealt with it.

Reply:
Ah, thanks for reminding me of what led me down this rabbit hole in the first place. I'll just restore the hash argument with `delete_if` to these client calls.

Reply:
Restored, thanks.